### ISSUE TITLE
Central Timezones generate incorrectly

### ISSUE THREAD

**Issue by andrewmc201** [moment/luxon#1038]
**Describe the bug**

Central timezones do not appear to calculate the correct offset in luxon 2.0.2

**To Reproduce**
Please share a minimal code example that triggers the problem:

`const date = DateTime.fromISO('2021-11-24T13:00:00.000-06:00').setZone('America/Chicago');`
`console.log(date.toFormat('hh a ZZZZ');`

**Actual vs Expected behavior**
We would expect the console log to output `01 PM CST`, but are getting `01 AM CST`
The server we are executing this on is set to Eastern Time. 

**Desktop (please complete the following information):**

- OS: CentOS 6
- Browser N/A (Node 10)
- Luxon version 2.0.2
- Your timezone "America/New_York" and "America/Chicago"

**Additional context**
Our server time is set to Eastern time, but the desired output is in Central time.

**Comment by icambron** [comment]
This doesn't reproduce for me in Node 16 where I get "01 PM CST".

This might have to do with the version of Node. Node 10 isn't supported in 2.x because it's EOLed, and I've removed code specifically designed to support it. I can't think offhand about what code that would be and how it would result in flipping AM and PM, but it's possibility,  so you might try a 1.x version and see what happens.

One thing to check is: is it the time or the formatting? In other works. if you call `toISO()` on `date` does it give you 13:00 or 1:00? My guess is that it's in formatting...

**Comment by andrewmc201** [comment]
Tried downgrading to a 1.23.x version and got the same result. If I console.log the actual date object, the output for both 2.x and 1.x shows the hour as 1, not as 13. Just to ensure that it wasn't an AM/PM issue, I set the time zone to both "America/Los_Angeles" as well as "America/New_York", and both of those actually gave me the correct hour values I would expect from their respective zones. In the case of "America/New_York", I saw an hour value of 14, and for "America/Los_Angeles", I saw an hour value of 11.
